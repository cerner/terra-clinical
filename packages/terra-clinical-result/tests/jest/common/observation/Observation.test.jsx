import React from 'react';
import Observation from '../../../../src/common/observation/_Observation';
import DefaultResult from '../../../../src/terra-dev-site/test/clinical-result/TestResults';

// Snapshot Tests
describe('Observation', () => {
  it('should render a default Observation', () => {
    const nameHeaderCell = shallow(<Observation />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  it('should render an Observation with eventId', () => {
    const nameHeaderCell = shallow(<Observation result={DefaultResult.result} eventId={DefaultResult.eventId} />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  it('should render an Observation with result', () => {
    const nameHeaderCell = shallow(<Observation result={DefaultResult.result} />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  it('should render an Observation with hideUnit', () => {
    const nameHeaderCell = shallow(<Observation result={DefaultResult.result} hideUnit />);
    expect(nameHeaderCell).toMatchSnapshot();
  });

  describe('- Interpretaion', () => {
    it('should render an Observation with interpretation critical', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="critical" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation critical-high', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="critical-high" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation critical-low', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="critical-low" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation positive', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="positive" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation abnormal', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="abnormal" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation high', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="high" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation low', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="low" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });

    it('should render an Observation with interpretation normal', () => {
      const nameHeaderCell = shallow(<Observation result={DefaultResult.result} interpretation="normal" />);
      expect(nameHeaderCell).toMatchSnapshot();
    });
  });
});
