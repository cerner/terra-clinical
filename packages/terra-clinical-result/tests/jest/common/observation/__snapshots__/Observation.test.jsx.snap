// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Observation - Interpretaion should render an Observation with interpretation abnormal 1`] = `
<Fragment>
  <span
    className="value abnormal"
  >
    <IconAbnormal
      className="icon-interpretation"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation - Interpretaion should render an Observation with interpretation critical 1`] = `
<Fragment>
  <span
    className="value critical"
  >
    <IconCritical
      className="icon-interpretation"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation - Interpretaion should render an Observation with interpretation critical-high 1`] = `
<Fragment>
  <span
    className="value critical"
  >
    <IconCritical
      className="icon-interpretation"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation - Interpretaion should render an Observation with interpretation critical-low 1`] = `
<Fragment>
  <span
    className="value critical"
  >
    <IconCritical
      className="icon-interpretation"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation - Interpretaion should render an Observation with interpretation high 1`] = `
<Fragment>
  <span
    className="value high"
  >
    <IconHigh
      className="icon-interpretation"
      data-name="Layer 1"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation - Interpretaion should render an Observation with interpretation low 1`] = `
<Fragment>
  <span
    className="value low"
  >
    <IconLow
      className="icon-interpretation"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation - Interpretaion should render an Observation with interpretation normal 1`] = `
<Fragment>
  <span
    className="value"
  >
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation - Interpretaion should render an Observation with interpretation positive 1`] = `
<Fragment>
  <span
    className="value positive"
  >
    <IconCritical
      className="icon-interpretation"
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation should render a default Observation 1`] = `<InjectIntl(ResultError) />`;

exports[`Observation should render an Observation with eventId 1`] = `
<Fragment>
  <span
    className="value"
  >
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation should render an Observation with hideUnit 1`] = `
<Fragment>
  <span
    className="value"
  >
    12345.678
  </span>
  <span
    className="unit unit-hidden"
  >
    mL
  </span>
</Fragment>
`;

exports[`Observation should render an Observation with result 1`] = `
<Fragment>
  <span
    className="value"
  >
    12345.678
  </span>
  <span
    className="unit"
  >
    mL
  </span>
</Fragment>
`;
